feature-dapp-impl/src/main/java/io/novafoundation/nova/feature_dapp_impl/presentation/browser/main/deeplink/DAppDeepLinkHandler.kt

```diff
@@ -36,7 +40,11 @@ class DAppDeepLinkHandler(
 
         ensureDAppInCatalog(normalizedUrl)
 
-        router.openDAppBrowser(DAppBrowserPayload.Address(url))
+        if (browserTabService.hasSelectedTab()) {
+            browserTabService.createAndSelectTab(normalizedUrl)
```

Comment:
Does this perform navigation to the browser? If not, seems like `router.openDAppBrowser(DAppBrowserPayload.Address(url))` should be moved out of `if` statement